db/migrations/0022.add_lti_type.py

```diff
@@ -0,0 +1,28 @@
+'''
+Add in the initial LTI Type Table
+'''
+
+from yoyo import step
+
+__depends__ = {'0020.add_lti_usage'}
+
+steps = [
+   step('''
+        CREATE TABLE IF NOT EXISTS lti_type (
+            canvas_id INTEGER NOT NULL,
+            des VARCHAR(200),
+            rpt_group_nm VARCHAR(200),
+            PRIMARY KEY (canvas_id)
+        ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4;
+    '''),
+    step('''
+        ALTER TABLE lti_placement MODIFY placement_type_id INTEGER NULL,
+        ADD FOREIGN KEY (placement_type_id) REFERENCES lti_type(canvas_id) 
+        ON DELETE SET NULL ON UPDATE CASCADE
```

Comment:
Question: Not sure I understand the rationale behind `SET NULL` versus `CASCADE`, since if you don't know the `lti_type`, the data becomes much less useful. *Maybe* this leaves the door open to a prior inventory of placements for a tool and then a reassignment of the tool ID once a tool's record is updated with new data. But we're dropping records and repopulating at the moment, so maintaining relationships like that isn't particularly relevant. Regardless, this is an edge case and I don't think it matters particularly, since we're not doing a bunch of one-off database transactions.

Reply:
Yeah, I didn't know what to do with this. It *probably* doesn't matter as you say right now but I also didn't want someone removing data to cause other data to remove.

I could just leave it as ON DELETE RESTRICT and not allow nulls. It sounds like this is the default so I really could just leave this out and not alter it to null. 

I guess I'll do that.

Reply:
Okay I updated this.